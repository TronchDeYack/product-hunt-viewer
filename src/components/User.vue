<template>
  <div class="user">
    <img :src="img" />
    <div class="content">
      <div class="name">
        {{ name }}
      </div>
      <span class="headline">
        {{ headline }}
      </span>
    </div>
  </div>
</template>

<style lang="scss" scoped>
.user {
  display: inline-flex;
  align-items: center;
  height: 50px;
  padding: 1rem;

  img {
    height: 100%;
    border-radius: 100%;
  }

  .content {
    text-align: left;
    margin-left: 15px;

    .name {
      font-weight: bold;
    }

    .headline {
      font-size: 12px;
    }
  }
}
</style>

<script>
export default {
  name: 'user',
  props: {
    name: String,
    headline: String,
    img: String,
  },
};
</script>
