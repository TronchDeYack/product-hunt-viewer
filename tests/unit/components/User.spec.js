import { shallowMount } from '@vue/test-utils';

import User from '@/components/User.vue';

describe('User', () => {
  it('Should render.', () => {
    const wrapper = shallowMount(User, {
      propsData: {
        name: 'Quos T.',
        img: 'https://ph-avatars.imgix.net/2250227/original?auto=format&fit=crop&crop=faces&w=original&h=original',
        headline: 'DevOps Engineer',
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
