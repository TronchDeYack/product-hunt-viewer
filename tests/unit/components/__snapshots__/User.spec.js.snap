// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Should render. 1`] = `
<div class="user"><img src="https://ph-avatars.imgix.net/2250227/original?auto=format&amp;fit=crop&amp;crop=faces&amp;w=original&amp;h=original">
  <div class="content">
    <div class="name">
      Quos T.
    </div> <span class="headline">
      DevOps Engineer
    </span>
  </div>
</div>
`;
